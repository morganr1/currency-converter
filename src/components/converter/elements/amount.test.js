import React from 'react';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';
import { Amount } from './Amount.element';

describe('Amount element', () => {
  const mockProps = {
    amount: 12,
    setAmountCallback: jest.fn(),
    resetForm: jest.fn(),
    buttonText: 'Test Button Text'
  };
  const wrapper = shallow(<Amount {...mockProps} />);

  it('renders correctly with the required props passed to it', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('onClick calls handleConversion function', () => {
    const btnComponent = wrapper.find(Button);
    btnComponent.simulate('click', { event: { target: { value: 2 } } });
    expect(mockProps.resetForm).toHaveBeenCalledTimes(1);
  });
});
