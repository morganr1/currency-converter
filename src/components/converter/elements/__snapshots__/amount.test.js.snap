// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount element renders correctly with the required props passed to it 1`] = `
<div
  className="makeStyles-container-2"
>
  <TextFieldComponent
    label="Amount"
    onChangeCallback={[Function]}
    value={12}
  />
  <WithStyles(ForwardRef(Button))
    aria-label="convert currency button"
    classes={
      Object {
        "root": "makeStyles-button-1",
      }
    }
    color="primary"
    onClick={[Function]}
    size="large"
    startIcon={<UNDEFINED />}
    variant="contained"
  >
    Test Button Text
  </WithStyles(ForwardRef(Button))>
</div>
`;
